src/core/api/ip6_api.cpp

```diff
@@ -132,8 +132,16 @@ void otIp6SetReceiveFilterEnabled(otInstance *aInstance, bool aEnabled)
 
 otError otIp6Send(otInstance *aInstance, otMessage *aMessage)
 {
-    return AsCoreType(aInstance).Get<Ip6::Ip6>().SendRaw(AsCoreType(aMessage),
-                                                         OPENTHREAD_CONFIG_IP6_ALLOW_LOOP_BACK_HOST_DATAGRAMS);
+    AsCoreType(aMessage).SetOrigin(OPENTHREAD_CONFIG_IP6_ALLOW_LOOP_BACK_HOST_DATAGRAMS
+                                       ? Message::Origin::kFromHostAllowLoopBack
+                                       : Message::Origin::kFromHostDisallowLoopBack);
+    return AsCoreType(aInstance).Get<Ip6::Ip6>().SendRaw(AsCoreType(aMessage));
+}
+
+otError otIp6SendFromOrigin(otInstance *aInstance, otMessage *aMessage, otMessageOrigin aOrigin)
+{
+    AsCoreType(aMessage).SetOrigin(static_cast<Message::Origin>(aOrigin));
+    return AsCoreType(aInstance).Get<Ip6::Ip6>().SendRaw(AsCoreType(aMessage));
 }
```

Comment:
Why not change `otIp6Send` to always send with `OT_MESSAGE_ORIGIN_FROM_HOST_UNTRUSTED`?

Are there still use cases that packets sent via `otIp6Send` can be trusted?

Reply:
Sorry I think my question isn't answered. Because I don't see more use cases of  `otIp6Send` without setting the origin to `OT_MESSAGE_ORIGIN_FROM_HOST_UNTRUSTED`.

Reply:
As Abtin suggested, we'll leave the API caller to decide what origin to use by `otMessageSetOrigin` instead of directly setting the origin in `otIp6Send`.